### PR TITLE
fix: pass req to ServerResponse to support pipeline() on node 16

### LINKED ISSUE #212: It cannot be used with pipeline()
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

unknown

### Plugin version

unknown

### Node.js version

v16.16.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

unknown

### Description

Under Node.js 16 ( v16.16.0 ), inject() cannot be used with pipeline() at the same time.

Under Node.js 14 ( v14.20.0 ), it works.

### Steps to Reproduce

1. Code.

```js
const { pipeline } = require("stream");

const { inject } = require("light-my-request");

(async () => {
  const resp = await inject((req, res) => {
    const data = `
  <!DOCTYPE html>
  <html>

    <head>
      <meta charset="UTF-8">
      <title>Document</title>
    </head>

    <body>
      <p>basic:ejs</p>
    </body>

  </html>
  `;

    res.statusCode = 200;

    pipeline(data, res, () => res.end());
  })
    .get("/")
    .end();

  console.log(resp.body);
})();
```

2. Run it.

```sh
  <!DOCTYPE html>
  <html>

    <head>
      <meta charset="UTF-8">
      <title>Document</title>
    </head>

    <body>
      <p>basic:ejs</p>
    </body>

  </html>
  
node:internal/streams/end-of-stream:206
    if (stream.req) stream.req.removeListener('finish', onfinish);
                               ^

TypeError: stream.req.removeListener is not a function
    at cleanup (node:internal/streams/end-of-stream:206:32)
    at pump (node:internal/streams/pipeline:166:5)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

### Expected Behavior

_No response_

### PR DESCRIPTION
**Pull request by xiedacon**
Close #212 

Pass the real req to ServerResponse, then it can be used with pipeline().

* https://github.com/nodejs/node/blob/96601cc759aa6f54f9cf9db466a004cb57ede651/lib/_http_server.js
* https://github.com/nodejs/node/blob/96601cc759aa6f54f9cf9db466a004cb57ede651/lib/internal/streams/end-of-stream.js

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/request.js [hunk 1]
```diff
@@ -92,9 +92,14 @@ function Request (options) {
...
 
+  /**
+   * https://github.com/nodejs/node/blob/75b68c6e4db515f76df73af476eccf382bbcb00a/lib/_http_common.js
+   */
   this.url = parsedURL.pathname + parsedURL.search
 
-  this.httpVersion = '1.1'
+  this.httpVersionMajor = 1
+  this.httpVersionMinor = 1
+  this.httpVersion = `${this.httpVersionMajor}.${this.httpVersionMinor}`
```
Comment (Uzlopak) [inline lib/request.js:102]:
```suggestion
  this.httpVersion = '1.1'
```

File: lib/request.js [hunk 2]
```diff
@@ -92,9 +92,14 @@ function Request (options) {
 
   const parsedURL = parseURL(options.url || options.path, options.query)
 
+  /**
+   * https://github.com/nodejs/node/blob/75b68c6e4db515f76df73af476eccf382bbcb00a/lib/_http_common.js
```
Comment (Uzlopak) [inline lib/request.js:96]: The link is not explaining anything. 

You mean this line?
https://github.com/nodejs/node/blob/75b68c6e4db515f76df73af476eccf382bbcb00a/lib/_http_common.js#L94

**Review by Uzlopak — CHANGES_REQUESTED** [review]

File: lib/request.js (hunk 2 above)
Comment (Uzlopak) [reply lib/request.js:96]: Maybe just remove the comment :/

File: lib/request.js (hunk 1 above)
Comment (xiedacon) [reply lib/request.js:102]: Is there any difference?

File: lib/request.js (hunk 2 above)
Comment (xiedacon) [reply lib/request.js:96]: ok

File: lib/request.js (hunk 1 above)
Comment (Uzlopak) [reply lib/request.js:102]: If you look at the code in https://github.com/nodejs/node/blob/75b68c6e4db515f76df73af476eccf382bbcb00a/lib/_http_common.js#L94 then you see that httpVersionMajor and httpVersionMinor are variable

But in our code here httpVersionMajor and httpVersionMinor are always set to 1. So we can save us some cycles to build the string and directly set it to `1.1`

Comment (xiedacon) [reply lib/request.js]: fixed

**Review by Uzlopak — APPROVED** [review]
LGTM

**Pull request merged**